### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by ajhunyady**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Format {
+    with: String,
+    using: Vec<String>,
+    output: String
+}
+
+/// Look-up values in json [ "/top/one", "/top/two"], and generate formatted string
+fn make_formatted_string(v: &Value, lookup: &Vec<String>, format_str: &String) -> Result<String> {
```
Comment (sehz) [inline src/lib.rs:49]: Since it is converting to Cow later, maybe better to accept Cow as input. Not show stopper

Comment (ajhunyady) [reply src/lib.rs:49]: Yeah, I'm not comfortable enough with Cow. It's something @galibey helped me with on a previous SmartModule. This is more or less a copy/paste of that.

File: test-data/transform.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  - uses: infinyon-labs/json-formatter@0.1.0
+    with:
+      spec:
+        match:
+          - key: "/type"
+            value: "subscribe"
+            format:
+              with: ":loudspeaker: {} ({}) subscribed on {}"
+              using: 
+                - "/name"
```
Comment (sehz) [inline test-data/transform.yaml:11]: why is `/` is needed?

Comment (ajhunyady) [reply test-data/transform.yaml:11]: `pointer` notation, it could also be "/user/info/name"

**Review by sehz — APPROVED** [review]
Looks good.  Question is about why need to `/` prefix.

**Comment by ajhunyady** [comment]
> Looks good. Question is about why need to `/` prefix.

replied inline.

File: test-data/transform.yaml (hunk 2 above)
Comment (ajhunyady) [reply test-data/transform.yaml:11]: https://www.rfc-editor.org/rfc/rfc6901

**Pull request merged**